### ISSUE TITLE
Inventory parameters for hosts excluded by --limit are not applied (edit: in delegation directive)

### ISSUE THREAD

**Issue by smangelsdorf** [ansible/ansible#8464]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.6.10 and 1.8
##### Environment:

CentOS 6.5 all around
##### Summary:

When running `ansible-playbook site.yml --limit appserver`, the `ansible_ssh_user` specified in the inventory for a different host is ignored, which breaks tasks that are `delegate_to: dbnode`
##### Steps To Reproduce:

_(This has all been stripped of my actual deployment details, but let me know if you need any more information)_

I have a task which looks like:

```
- name: 'Create database'
  mysql_db: >
    login_unix_socket=/var/run/mysqld/mysqld.sock name={{ app.db.name }}
    collation=utf8_general_ci encoding=utf8 state=present
  delegate_to: '{{ app.db.node }}'
```

Inventory looks like:

```
[approle]
appserver ansible_ssh_user=root

[databases]
dbnode ansible_ssh_user=root
```

And the command line was:

```
ansible-playbook site.yml --limit approle
```
##### Expected Results:

I expected that when connecting to the `dbnode` host, the `root` user would be used for authentication as specified in the inventory.
##### Actual Results:

When I run the playbook, it fails on this task with:

```
TASK: [app | Create database] *****************
<dbnode> ESTABLISH CONNECTION FOR USER: ansible on PORT 22 TO dbnode
fatal: [appserver] => {'msg': 'FAILED: Authentication failed.', 'failed': True}
```

I can work around this (and in my case I am happy to keep this as a permanent fix) by adding the option to `ansible.cfg`:

```
[defaults]
remote_user = root
```

**Comment by srgvg** [comment]
There have been quite some changes lately related to delegation hosts, those special vars, and inventory in general.

Could you tests this with the development tree? In 1.7 variables get loaded for all hosts, also those outside the --limit.

**Comment by smangelsdorf** [comment]
I've managed to reproduce this in the latest `devel` commit:

```
$ ansible --version
ansible 1.8 (devel 2d54448064) last updated 2014/08/07 10:24:29 (GMT +1000)
```

Without `--limit`:

```
$ ANSIBLE_NOCOWS=1 ansible-playbook -i development test.yml

PLAY [all] ********************************************************************

GATHERING FACTS ***************************************************************
ok: [app4.macaroon.cc.st]
ok: [db1.macaroon.cc.st]

TASK: [ping] ******************************************************************
ok: [db1.macaroon.cc.st -> db1.macaroon.cc.st]
ok: [app4.macaroon.cc.st -> db1.macaroon.cc.st]

PLAY RECAP ********************************************************************
app4.macaroon.cc.st        : ok=2    changed=0    unreachable=0    failed=0
db1.macaroon.cc.st         : ok=2    changed=0    unreachable=0    failed=0
```

When using `--limit`:

```
$ ANSIBLE_NOCOWS=1 ansible-playbook -vvv -i development test.yml --limit app4.macaroon.cc.st

PLAY [all] ********************************************************************

GATHERING FACTS ***************************************************************
[... snipped debug output ...]
ok: [app4.macaroon.cc.st]

TASK: [ping] ******************************************************************
<db1.macaroon.cc.st> ESTABLISH CONNECTION FOR USER: shaun
<db1.macaroon.cc.st> REMOTE_MODULE ping
<db1.macaroon.cc.st> EXEC ['ssh', '-C', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/Users/shaun/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'ConnectTimeout=10', 'db1.macaroon.cc.st', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1407371663.28-242512219923899 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1407371663.28-242512219923899 && echo $HOME/.ansible/tmp/ansible-tmp-1407371663.28-242512219923899'"]
fatal: [app4.macaroon.cc.st -> db1.macaroon.cc.st] => SSH encountered an unknown error during the connection. We recommend you re-run the command using -vvvv, which will enable SSH debugging output to help diagnose the issue

FATAL: all hosts have already failed -- aborting

PLAY RECAP ********************************************************************
           to retry, use: --limit @/Users/shaun/test.retry

app4.macaroon.cc.st        : ok=1    changed=0    unreachable=1    failed=0
```

Inventory file:

```
[development]
app4.macaroon.cc.st ansible_ssh_user=root
db1.macaroon.cc.st ansible_ssh_user=root
```

Playbook:

```
---
- hosts: all
  tasks:
    - action: ping
      delegate_to: db1.macaroon.cc.st
```

**Comment by bcoca** [comment]
fixed in current devel via #15024
